File: 프로그래머스/3/12904. 가장 긴 팰린드롬/가장 긴 팰린드롬.java
import java.util.*;

class Solution
{
    public int solution(String s)
    {
        int answer = 1;
        int N = s.length();
        int palindrome[][] = new int[N][N];
        
        for(int i=0; i<N; i++){
            for(int j=0; j<i+1; j++){
                if(s.charAt(i) != s.charAt(j)) continue;
                if(Math.abs(i - j) > 1 && palindrome[i-1][j+1] != 1) continue;
                palindrome[i][j] = 1;
                answer = Math.max(answer, i-j+1);
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/3/12904. 가장 긴 팰린드롬/가장 긴 팰린드롬.js
function solution(s)
{
    var answer = 1;
    var N = s.length;
    const palindrome = Array.from({length : N}, () => Array(N).fill(0));
    
    for(let i=0; i<N; i++){
        for(let j=0; j<i+1; j++){
            if(s.charAt(i) != s.charAt(j)) continue;
            if(i-j > 1 && palindrome[i-1][j+1] != 1) continue;
            palindrome[i][j] = 1;
            answer = Math.max(answer, i-j+1);
        }
    }

    return answer;
}